src/celeritas/inp/Field.hh

```diff
@@ -44,6 +47,9 @@ struct UniformField
 
     //! Field driver options
     FieldDriverOptions driver_options;
+
+    //! Volumes where the field is present (optional)
+    std::vector<VolumeId> volumes;
```

Comment:
Volume ID isn't really a safe way to do this because (as currently implemented) the IDs change based on the implementation (or even by switching the order of volumes in a file).

The equivalent for SDs uses a variant set:
```c++
    using SetVolume = std::unordered_set<G4LogicalVolume const*>;
    using SetString = std::unordered_set<std::string>;
    using VariantSetVolume = std::variant<SetVolume, SetString>;
```
and since "no volumes" really means "all volumes" I'd say it should default to "monostate", which is the standard way of making a variant optional:
```c++
    using VariantSetVolume = std::variant<std::monostate, SetVolume, SetString>;
```

Let's implement this change after this PR is merged (either merging main into #1632 or as a separate PR).

Reply:
Nice! I was wondering about a clean way to provide either LVs or names. Sounds good, I'll hold off on updating this.